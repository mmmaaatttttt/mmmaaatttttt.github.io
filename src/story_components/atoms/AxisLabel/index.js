import styled from "styled-components";
import PropTypes from "prop-types";

const AxisLabel = styled.text`
  text-anchor: ${props => props.anchor};
  font-style: italic;
  font-size: 125%;
  font-weight: bold;
  stroke: #fff;
`;

AxisLabel.propTypes = {
  anchor: PropTypes.oneOf(["start", "middle", "end"]).isRequired
};

AxisLabel.defaultProps = {
  anchor: "middle"
};

export default AxisLabel;
