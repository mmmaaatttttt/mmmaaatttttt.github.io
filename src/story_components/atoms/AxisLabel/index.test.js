import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import AxisLabel from ".";

const wrap = (props = {}, text = "") =>
  shallow(<AxisLabel {...props}>{text}</AxisLabel>);

it("renders successfully", () => {
  const wrapper = wrap();
  expect(toJson(wrapper)).toMatchSnapshot();
});

it("renders text when passed in successfully", () => {
  const wrapper = wrap({}, "x-axis");
  expect(toJson(wrapper)).toMatchSnapshot();
});

it("renders props when passed in", () => {
  ["start", "middle", "end"].forEach(anchor => {
    const wrapper = wrap({ anchor }, "axis label");
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
