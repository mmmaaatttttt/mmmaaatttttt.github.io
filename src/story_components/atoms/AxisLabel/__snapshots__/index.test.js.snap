// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders props when passed in 1`] = `
<text
  className="sc-bdVaJa iCJtsk"
>
  axis label
</text>
`;

exports[`renders props when passed in 2`] = `
<text
  className="sc-bdVaJa fFNPsM"
>
  axis label
</text>
`;

exports[`renders props when passed in 3`] = `
<text
  className="sc-bdVaJa jWwhba"
>
  axis label
</text>
`;

exports[`renders successfully 1`] = `
<text
  className="sc-bdVaJa fFNPsM"
/>
`;

exports[`renders text when passed in successfully 1`] = `
<text
  className="sc-bdVaJa fFNPsM"
>
  x-axis
</text>
`;
